### PR TITLE
Append the transcript lines to the DOM.

### PR DESCRIPTION
**Pull request by amuamushu**
Transcript line text and start time now appear on DOM. In a later pull request, the format of each line will be updated to look like the transcript diagram in the [design document](https://docs.google.com/document/d/12t7FDPDpTpHzrQBJ3T8qdmGKGrPi3lr2-UPmoc2ISHI/edit#heading=h.5irk4csrpu0y).

Transcript on the lecture-view page with the discussion: https://screenshot.googleplex.com/SRZ3Ze9w9f2.png

### CODE REVIEW

**Comment by amuamushu** [comment]
@googlebot I fixed it.

**Comment by amuamushu** [comment]
I haven't resolved the merge conflicts yet because when I tried to it deleted most of the changes I've made so far. I'm hoping that when I merge this branch into master later, there won't be this issue.

Screenshots of issue: https://screenshot.googleplex.com/3QDTmFBL4AG.png
https://screenshot.googleplex.com/ai0LmuLrRpz.png

File: src/main/webapp/transcript.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (jaysaleh) [inline src/main/webapp/transcript.js:1]: Add file comments

File: src/main/webapp/transcript.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const TRANSCRIPT_CONTAINER = 'transcript-container';
+
+/**
+ * Sends a POST request to the transcript.
+ */
+function sendPostToTranscript() {
+  // TODO: Update with user input.
+  const paramsString = 'id=123456789&video=3ymwOvzhwHs';
+  const params = new URLSearchParams(paramsString);
+  fetch('/transcript', {method: 'POST', body: params})
```
Comment (jaysaleh) [inline src/main/webapp/transcript.js:11]: Maybe make '/transcript' a const

File: src/main/webapp/transcript.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Fetches the transcript lines from \transcript.
+ *
+ * <p>{@code paramsString} indicates the video ID
+ * to fetch the transcript from.
+ */
+function fetchTranscriptLines(paramsString) {
+  fetch(
+      '/transcript' +
```
Comment (jaysaleh) [inline src/main/webapp/transcript.js:23]: Add const for /transcript and maybe ?

File: src/main/webapp/transcript.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+function sendPostToTranscript() {
+  // TODO: Update with user input.
+  const paramsString = 'id=123456789&video=3ymwOvzhwHs';
+  const params = new URLSearchParams(paramsString);
+  fetch('/transcript', {method: 'POST', body: params})
+      .then(fetchTranscriptLines(paramsString));
+}
+
+/**
+ * Fetches the transcript lines from \transcript.
```
Comment (jaysaleh) [inline src/main/webapp/transcript.js:16]: /transcript

File: src/main/webapp/transcript.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Creates a <p> tag to store the given {@code text} inside the
+ * {@code container} and returns the <p> tag using the given text.
+ */
+function appendPTagToContainer(text, container) {
+  const pTag = document.createElement('p');
+  pTag.innerText = text;
+  container.appendChild(pTag);
+  return pTag;
```
Comment (jaysaleh) [inline src/main/webapp/transcript.js:70]: Is there a reason why this is returned, I don't see the return being used in the caller

File: src/main/webapp/transcript.js (hunk 2 above)
Comment (n-wach) [reply src/main/webapp/transcript.js:11]: `ENDPOINT_TRANSCRIPT`

File: src/main/webapp/transcript.js (hunk 4 above)
Comment (n-wach) [reply src/main/webapp/transcript.js:16]: Either `@link` to the servlet or `@code` to the constant for the url

File: src/main/webapp/transcript.js (hunk 3 above)
Comment (n-wach) [reply src/main/webapp/transcript.js:23]: Alternatively use UrlBuilder, but it looks like this is using hardcoded test param string?

**Comment by n-wach** [comment]
Add comments to Jay's comments

File: src/main/webapp/transcript.js (hunk 1 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:1]: done!

File: src/main/webapp/transcript.js (hunk 2 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:11]: done!

File: src/main/webapp/transcript.js (hunk 4 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:16]: Thanks for catching that! Added @code for `ENDPOINT_TRANSCRIPT`

File: src/main/webapp/transcript.js (hunk 3 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:23]: I think the question mark will be removed when I use the actual URL from your branch. 

I made an issue for converting to Url Buider later #42

File: src/main/webapp/transcript.js (hunk 5 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:70]: I returned it so I can update the classes for the p tag in a later pull request.

I updated the method name to be `appendAndReturnPTagToContainer` to make it more clear that the p tag is returned. Thanks!

File: src/main/webapp/transcript.js (hunk 3 above)
Comment (n-wach) [reply src/main/webapp/transcript.js]: Sounds good, maybe add a TODO?

File: src/main/webapp/transcript.js (hunk 5 above)
Comment (jaysaleh) [reply src/main/webapp/transcript.js:83]: I don't think we need to put return in the method name, reader could just look for a return statement

File: src/main/webapp/transcript.js (hunk 3 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js]: Thanks for reminding me! Done

File: src/main/webapp/transcript.js (hunk 5 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js:83]: Good point. Removed.

File: src/main/webapp/index.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+<title>ZoomTube</title>
+<link rel="stylesheet" href="style.css" />
+<script src="transcript.js"></script>
+<body>
+  <p>Hello world!</p>
+  <!-- TODO: Replace Button with a form that uses User Input. -->
+  <button type="button" onclick="sendPostToTranscript();">sendPostToTranscript</button>
+  <div id="transcript-container"></div>
+</body>
```
Comment (n-wach) [inline src/main/webapp/index.html:12]: is index.html just for testing?  Can this be merged with lecture-view.html?

Comment (amuamushu) [reply src/main/webapp/index.html]: Yes, that is just for testing. Thank you for reminding me to move it over.
Done!

File: src/main/webapp/transcript.js [hunk 7]
```diff
@@ -1,9 +1,84 @@
...
 /**
  * Sends a POST request to the transcript.
  */
 function sendPostToTranscript() {
-  // TODO: Update with user input.
-  const testParamsString = 'id=123456789&video=8PrOp9t0PyQ';
-  const params = new URLSearchParams(testParamsString);
-  fetch('/transcript', {method: 'POST', body: params});
+  // TODO: Update with user input and URL Builder.
+  const paramsString = 'id=123456789&video=3ymwOvzhwHs';
```
Comment (billie-cat) [inline src/main/webapp/transcript.js:24]: paramsString is not very descriptive or what this id is for? Is it transcriptUrl?

File: src/main/webapp/transcript.js [hunk 8]
```diff
@@ -1,9 +1,84 @@
...
+  const params = new URLSearchParams(paramsString);
+  fetch(ENDPOINT_TRANSCRIPT, {method: 'POST', body: params})
+      .then(fetchTranscriptLines(paramsString));
+}
+
+/**
+ * Fetches the transcript lines from {@code ENDPOINT_TRANSCRIPT}.
+ *
+ * <p>{@code paramsString} indicates the video ID
```
Comment (billie-cat) [inline src/main/webapp/transcript.js:33]: Why not rename paramsString to videoId. Seems more descriptive. paramsString does not indicate much about what it is.

File: src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java [hunk 9]
```diff
@@ -1,224 +0,0 @@
-// Copyright 2019 Google LLC
```
Comment (billie-cat) [inline src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java:1]: Why is this file deleted?

Comment (amuamushu) [reply src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java:1]: This file was originally in the branch that I'm merging to but after resolving my merge conflict, it got added over to this branch. The file still exists in transcript-store.

File: src/main/webapp/transcript.js (hunk 8 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js]: I forgot to mention that the string also includes the lecture ID. I updated the param description to reflect this.

File: src/main/webapp/transcript.js (hunk 7 above)
Comment (amuamushu) [reply src/main/webapp/transcript.js]: The string is the query parameters for a certain lecture recording. I updated the variable name to be `lectureQueryString` so hopefully it's clearer.

**Comment by amuamushu** [comment]
> LGTM.
> I just one small question about the screen shot. It seems transcript lines are repeated in the first one but not the second one, did the ui change?

Thank you for catching that! The second one was meant for the next pull request. The UI is supposed to look like what was in the first pull request. The lines were repeated in the first one because the datastore wasn't cleared beforehand. 

I updated the screenshot with what it looks like after clearing the datastore. Sorry for the confusion!

File: src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java (hunk 9 above)
Comment (billie-cat) [reply src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java:1]: Sounds good!

File: src/main/webapp/transcript.js [hunk 10]
```diff
@@ -1,9 +1,84 @@
...
+
+/**
+ * Creates a <p> tag to store the given {@code text} inside the
+ * {@code container} and returns the <p> tag using the given text.
+ */
+function appendPTagToContainer(text, container) {
+  const pTag = document.createElement('p');
+  pTag.innerText = text;
+  container.appendChild(pTag);
+  return pTag;
```
Comment (billie-cat) [inline src/main/webapp/transcript.js:83]: Why are you adding it to the container and returning it? Typically you do one or the other.

Comment (amuamushu) [reply src/main/webapp/transcript.js:86]: I returned it so that I can add classes to the p tag in a different method. I'm planning on adding the classes in a subsequent pull request.

I updated the method header to reflect this fact. Thanks!

File: src/main/webapp/transcript.js [hunk 11]
```diff
@@ -1,9 +1,87 @@
...
+  appendPTagToContainer(transcriptLine.content, liElement);
+  liElement.appendChild(document.createElement('hr'));
+  ulElement.appendChild(liElement);
+}
+
+/**
+ * Creates a <p> tag to store the given {@code text} inside the
+ * {@code container} and returns the <p> tag using the given text
+ * 
+ * <p>The <p> tag is returned so that the calling method can add additional
```
Comment (billie-cat) [inline src/main/webapp/transcript.js:79]: I just realized you might need to add backslashes \<p\> or else it will add a new paragraph :/ Maybe create a tracking bug for this?

Comment (amuamushu) [reply src/main/webapp/transcript.js:79]: I created an issue for this #99. Thanks for noticing that!
